File: C#/Services/WeatherService.cs
namespace WeatherAPI.Services;

using System.Net.Http;
using System.Threading.Tasks;
using Newtonsoft.Json;

public class WeatherService
{
    private readonly HttpClient _httpClient;
    private string? API_URL = Environment.GetEnvironmentVariable("API_URL");
    private string? API_KEY = Environment.GetEnvironmentVariable("API_KEY");
    
    public WeatherService(HttpClient httpClient)
    {
        _httpClient = httpClient;
    }

    public async Task<WeatherData?> GetWeatherDataAsync(double latitude, double longitude)
    {
        Console.WriteLine(API_URL + "\n" + API_KEY + "\n" + latitude + "\n" + longitude);
        string apiUrl = $"{API_URL}?lat={latitude}&lon={longitude}&appid={API_KEY}&units=metric";
        HttpResponseMessage response = await _httpClient.GetAsync(apiUrl);

        string jsonResponse = await response.Content.ReadAsStringAsync();
        var weatherData = JsonConvert .DeserializeObject<WeatherData>(jsonResponse);
        return weatherData ?? null;
    }
}

File: src/main/java/com/example/weatherapijava/service/WeatherService.java
package com.example.weatherapijava.service;

import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

import com.example.weatherapijava.model.WeatherData;

@Service
public class WeatherService {

    private final String API_URL = System.getenv("API_URL");
    private final String API_KEY = System.getenv("API_KEY");

    private final RestTemplate restTemplate;

    public WeatherService(RestTemplate restTemplate) {
        this.restTemplate = restTemplate;
    }

    public WeatherData getWeatherData(double latitude, double longitude) {
        String apiUrl = String.format("%s?lat=%s&lon=%s&appid=%s", API_URL, latitude, longitude, API_KEY);
        return restTemplate.getForObject(apiUrl, WeatherData.class);
    }
}
